kalite/main/api_views.py

```diff
@@ -254,40 +253,29 @@ def exercise(request, ds, exercise_id):
     exercise = get_exercise_data(request, exercise_id)
     if "facility_user" in request.session:
         facility_id = request.session["facility_user"].facility.id
-        current_unit = get_current_unit_settings_value(facility_id)
         student_grade = ds["ab_testing"].student_grade_level
         if student_grade:
-            if not UNIT_EXERCISES.has_key(current_unit):
-                for playlist in Playlist.all():
-                    if not UNIT_EXERCISES.has_key(playlist.unit):
-                        UNIT_EXERCISES[playlist.unit] = {}
-                    # Assumes gx_pyy id for playlist.
-                    grade = int(playlist.id[1])
-                    if not UNIT_EXERCISES[playlist.unit].has_key(grade):
-                        UNIT_EXERCISES[playlist.unit][grade] = []
-                    for entry in playlist.entries:
-                        if entry["entity_kind"] == "Exercise":
-                            UNIT_EXERCISES[playlist.unit][grade].append(entry["entity_id"])
-
-            current_unit_exercises = UNIT_EXERCISES.get(current_unit, {}).get(student_grade, [])
+            for playlist in Playlist.all():
```

Comment:
Can also cut out all of this looping, as we shouldn't care any more about how many exercises there are.